### DISCUSSION TITLE
关于访问ConfigServer总是以IP形式访问

### DISCUSSION THREAD

**Discussion by cyenxchen** [alibaba/ilogtail#942] [Help] [answered]
ilogtail配置如下，正常情况下，当ilgotail调用libcurl库向ConfigServer发送请求之前总会先将域名解析成IP，也就是将example.com解析成对应的IP，这时候如果这个域名是ingress就会有问题
```json
{                                        
  "ilogtail_configserver_address" : [    
    "example.com"           
  ]
}      
``` 
关键代码如下图：
<img width="887" alt="image" src="https://github.com/alibaba/ilogtail/assets/22564041/82fef02e-0911-4157-879b-db04170199c5"> 
在没有配置代理的情况下，`IsHostIPReplacePolicyEnabled()`总是返回默认值true导致解析域名为IP。我现在做法是`IsHostIPReplacePolicyEnabled()` 函数返回的成员变量在初始化是设置为false解决：
```c++
    bool mEnableHostIPReplace = false;
```
我的想法是在调用libcurl之前往HTTP header塞一个Host header，值就是ConfigServer的域名？或者向我上面那样将默认值设置为false？（如果是这样的话那么代码里的dns缓存功能就没用了） 又或者增加一个配置选项标识ConfigServer的域名指向的是ingress，然后根据这个配置来选择是否解析成IP？

**Comment by messixukejia** [comment]
@Takuka0311 帮忙看下

**Comment by Takuka0311** [comment]
请问“域名是ingress就会有问题”具体是怎样的问题呢？有没有报错信息之类的

**Reply by cyenxchen** [reply]
ingress控制器是靠http header中的Host来匹配的，而这个Host一般都是设置成域名的，直接使用IP发送请求HTTP header的Host就是IP，无法匹配ingress规则，从而导致访问404。

**Reply by Takuka0311** [reply, marked as answer]
这个参数是可以通过config或者env配置的，感觉并不需要修改代码
![image](https://github.com/alibaba/ilogtail/assets/66948919/049e2c62-3fd3-47d4-82fc-0fbf431d3d08)

**Reply by cyenxchen** [reply]
刚试了下环境变量确实可以，是我之前看漏代码了，感谢。
